src/affineNormL2.jl

```diff
@@ -0,0 +1,74 @@
+# Composition of the L2 norm with an affine function
+export AffineNormL2
+
+@doc raw"""
+    AffineNormL2(A,b,λ=1)
+
+Returns the ``\ell_{2}`` norm operator composed with an affine function
+```math
+f(x) = λ ||Ax+b||_2
+```
+where ``\lambda > 0``, A is some mxn matrix and b is a vector in Rᵐ.
+"""
+struct AffineNormL2{R <: Real, V1 <: AbstractMatrix{R}, V2 <: AbstractVector{R}}
+  lambda::R
+  A::V1
+  b::V2
+  function AffineNormL2{R,V1,V2}(A::V1,b::V2,lambda::R) where {R <: Real, V1 <: AbstractMatrix{R}, V2 <: AbstractVector{R}}
+    if lambda < 0
+      error("Affine Norm L2 : parameter λ must be nonnegative")
+    elseif size(A,1) != length(b)
+      error("Affine Norm L2 : dimensions of parameter A and b must match")
```

Comment:
Also, we must have `size(A, 1) ≤ size(A, 2)`, right? Otherwise, there is no chance that `A` would have full row rank.